### ISSUE TITLE
Nomos regexp error

### ISSUE THREAD

**Issue by AMDmi3** [fossology/fossology#754]
There seem to be problem with specific nomos regexp:

```
Compile failed, regex #706
regex = "program is linked with (|and uses) microsoft distributable code"
FATAL nomos_regex.c.49: regcomp failure: empty (sub)expression
```

**Comment by mcjaeger** [comment]
thanks for pointing out, in order to cover it could you please help and provide us with more details when you have encountered this error?

**Comment by AMDmi3** [comment]
FreeBSD 11.0 amd64, nomos standalone build, pcre 8.4. The problem is reproducible when running nomos on a text file with single line:

```
program is linked with microsoft distributable code
```

I think the problem is obvious from regexp definition. There also seem to be a double space problem (e.g. the rule matches ```with<space>and<space>uses<space>microsoft``` and ```with<space><space>microsoft```, but not ```with<space>microsoft```)

The problems is fixed for me by

```
-%STR%  "program is linked with (|and uses) microsoft distributable code"
+%STR%  "program is linked with( | and uses )microsoft distributable code"
```

**Comment by mcjaeger** [comment]
actually, for the examples above it is fixed:

https://github.com/fossology/fossology/pull/817

but apparently another similar case has been found:

https://github.com/fossology/fossology/issues/822

**Comment by AMDmi3** [comment]
> actually, for the examples above it is fixed

It isn't. After the mentioned change there still is an empty subexpression.

**Comment by max-wittig** [comment]
but didn't you open #822 for the same issue? Or are these different issues?

**Comment by AMDmi3** [comment]
The same as I've originally submitted, but not the same that was fixed here in 0b8e58b.

**Comment by max-wittig** [comment]
I get that. But is it correct to close one if the issues you've created for this problem?

**Comment by AMDmi3** [comment]
Yes, let's leave this one closed and #822 active, with additional examples.
